src/gradient.jl

```diff
@@ -80,13 +80,13 @@ function extract_gradient!(::Type{T}, result::DiffResult, dual::Dual) where {T}
 end
 
 extract_gradient!(::Type{T}, result::AbstractArray, y::Real) where {T} = fill!(result, zero(y))
-extract_gradient!(::Type{T}, result::AbstractArray, dual::Dual) where {T}= copyto!(result, partials(T, dual))
+function extract_gradient!(::Type{T}, result::AbstractArray, dual::Dual) where {T}
+    result[:] .= partials(T, dual)
+    return result
+end
 
 function extract_gradient_chunk!(::Type{T}, result, dual, index, chunksize) where {T}
-    offset = index - 1
-    for i in 1:chunksize
-        result[i + offset] = partials(T, dual, i)
-    end
+    result[index:index+chunksize-1] .= partials.(T, dual, 1:chunksize)
```

Comment:
Similarly, maybe one approach here would be to use something like
```suggestion
    copyto!(result, index, partials(T, dual).values, 1, chunksize)
```
Again that would require that CUDA/GPUArrays defines `copyto!` for `Tuple`s.

Reply:
JLArrays seems to have it but my GPU cluster is down atm to double check with CUDA, but this is promising!